### DISCUSSION TITLE
Shift + Cursor Keys should also mark the last selected file

### DISCUSSION THREAD

**Discussion by sagamusix** [doublecmd/doublecmd#1173] [Suggestions]
I understand that the current behaviour is probably due to how TotalCommander does things, and maybe there's even an option to turn it off that I haven't found yet, but the following behaviour annoys me somewhat especially since it isn't present in other programs also following the orthodox file manager pattern (e.g. WinSCP): When I select a file and then extend the selection using Shift+Up/Down, the last selected file is not part of the selection. However, if I draw exactly the same selection by Shift-Clicking on the last file, then it is part of the selection. I'd greatly appreciate this also was the case when using Shift+Cursor Keys.

**Comment by WhiteChairFromIkea** [comment]
I agree, there must be an option for that. Now, when I have learned how it works there is no problem anymore, but this behavior is a black sheep among others, need to refink my workflow in regular explorers and different computers..

Also, this was one of the first issues I have reported in unofficial repo, when started using dc 😁. There are several comments there: https://github.com/double-commander/doublecmd/issues/13

**Comment by goyalyashpal** [comment]
_**Edit:** uh oh, sorry, i ..._

~This issue can be closed as a duplicate of above mentioned double-commander/doublecmd#13~

**Comment by goyalyashpal** [comment]
- https://github.com/double-commander/doublecmd/issues/33

oh - uh oh, i _[AGAIN]_ missed the details about repos - what's happening to me. goshhhhh. sorry.

double-commander vs doublecmd

[again]: https://github.com/doublecmd/doublecmd.github.io/issues/52#issuecomment-1427072782

**Comment by goyalyashpal** [comment]
_moving over my input VERBATIM from over there_

----
> Now, when I have learned how it works there is no problem anymore ...
\- @ WhiteChairFromIkea on 2021.12.xx at https://github.com/doublecmd/doublecmd/issues/281

the correct way to select files in doublecmd is to use `space bar` and arrow keys. it works wonders once u leave behind the windows explorer's legacy - which u already have 😉 

Some notes: 
- instead of the normal arrow keys, u can also use the numpad-navigation keys (i.e. when numlock off) to perform same task.
- so:
- `shift + down arrow` work the same as pressing `Insert` or `shift + num2 (numlock off)`
- `shift + up arrow` work same as pressing `shift + num8 (numlock off)`

This is slightly better than the `space` + `arrow` as it brings down the keypress count from 2 every time to just 1 with modifier pressed throughout. However, i have one suggestion which _i think_ will improve ur use case further.

**Comment by sagamusix** [comment]
Thanks but no thanks. I have already outlined in my initial post why I disagree with Double Commander's behaviour and why I think there should be an option to toggle between WinSCP/Explorer/etc.-like behaviour and TotalCommander-like behaviour.

**Comment by goyalyashpal** [comment]
Just to be clear, so, u are asking for _only the proposed solution_ i.e. the underlined problem is not the main concern for u, right?

**Comment by WhiteChairFromIkea** [comment]
> Just to be clear, so, u are asking for _only the proposed solution_ i.e. the underlined problem is not the main concern for u, right?

The point is that very odd (for win explorer user) out of the box experience is frightening newcomers. The option to switch between explorer like and orthodox like behaviour would be very nice, this is a valid feature request. Since this is an open source project, maybe someone will implement it. Host of this project is really friendly and accepts pull requests. Btw, thanks for tips @goyalyashpal, it is always welcome when trying to help 🙂

Even created this ticket for helping newcomers: https://github.com/doublecmd/doublecmd/issues/167

**Comment by goyalyashpal** [comment]
> ... odd OOBE is frightening newcomers

i am as newcomer to orthodox fule managers as one can get - started using it since just 4 weeks ago - no prior experience anywhere, liked it immediately, have never used windows explorer since. So, no! this point doesnt hold, u gotta be more specific than just "newcomers". But anyways.

> this is an open source project, maybe someone will implement it

yes, that's why i asked if this is a feature request (from a reference software) or aimed at the problem (of last line selection)

i am not discouraging anyone, i just want to know the scope of the issue.

**Comment by sagamusix** [comment]
As said, I would appreciate if there was *an option* to toggle this behaviour - hence this is a feature request.

**Comment by goyalyashpal** [comment]
ohkay, good.

**Comment by popy2k14** [comment]
Have to work on different systems (linux with currently dolphin/nemo, windows for work, ...),
and when using double commander, i always have to rethink how it works (resulting in missing selected files for operations...).

+1 for an explorer/dolphin like selection mode.
The seperate (cursor and selection) mode feels off to me.
